packages/netlify-cms-core/src/components/Editor/EditorToolbar.js

```diff
@@ -393,7 +393,7 @@ class EditorToolbar extends React.Component {
           </PublishedToolbarButton>
         )}
       >
-        {canPublish && (
+        {(!canNotDelete && canPublish) && (
```

Comment:
Can we rename this to `canDelete` and avoid the negation?

I think `const canDelete = collection.get('delete', true)` should work